client/network/src/protocol.rs

```diff
@@ -653,10 +653,14 @@ impl<B: BlockT> Protocol<B> {
 				} else {
 					None
 				},
-				justification: if !block_data.justification.is_empty() {
-					Some(block_data.justification)
+				justifications: if !block_data.justification.is_empty() {
+					// For compatibility we assume that the incoming Justifications is an
+					// `EncodedJustification`, as before, and that it can only be for GRANDPA.
+					let justification: EncodedJustification = block_data.justification;
+					let justification = (GRANDPA_ENGINE_ID, justification);
```

Comment:
I'd be in favour of just doing this (provided it's explained in a comment), and not polluting `lib.rs` with a constant.

```suggestion
					let justification = (b"FRNK", justification);
```